### PR TITLE
split config/configauth module

### PR DESCRIPTION
**Pull request by codeboten**
Linked issue #7895

Follow up to #7908

### CODE REVIEW

File: component/go.mod [hunk 1]
```diff
@@ -35,6 +35,8 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
+
+
```
Comment (jpkrohling) [inline component/go.mod:39]: Was this added by a tool?

File: config/configauth/go.mod [hunk 2]
```diff
@@ -0,0 +1,54 @@
+module go.opentelemetry.io/collector/config/configauth
+
+go 1.19
+
+
+
```
Comment (jpkrohling) [inline config/configauth/go.mod:6]: This go.mod looks strange now: perhaps add the direct dependencies first, the transitive after that, and the overrides at the bottom?

File: extension/auth/go.mod [hunk 3]
```diff
@@ -36,38 +36,38 @@ require (
...
-replace go.opentelemetry.io/collector/config/configcompression => ../../config/configcompression
 
-replace go.opentelemetry.io/collector/config/confignet => ../../config/confignet
 
-replace go.opentelemetry.io/collector/config/configopaque => ../../config/configopaque
 
-replace go.opentelemetry.io/collector/config/configtls => ../../config/configtls
+
+
+
```
Comment (jpkrohling) [inline extension/auth/go.mod:66]: Can you check the empty lines here?

File: extension/go.mod [hunk 4]
```diff
@@ -35,38 +35,38 @@ require (
...
 
-replace go.opentelemetry.io/collector/config/confignet => ../config/confignet
 
-replace go.opentelemetry.io/collector/config/configopaque => ../config/configopaque
 
-replace go.opentelemetry.io/collector/config/configtls => ../config/configtls
+
+
+
+
```
Comment (jpkrohling) [inline extension/go.mod:65]: And here.

**Review by jpkrohling — COMMENTED** [review]
Apparently, there are some formatting issues. Nothing that would block this PR from being merged, but nice to have them fixed anyway. If you prefer to fix them at a later time, let me know and I'll approve the PR.

File: component/go.mod (hunk 1 above)
Comment (codeboten) [reply component/go.mod]: this was added in the previous PR by crosslink removing other dependencies and leaving the new lines until i re-ran the tool

File: config/configauth/go.mod (hunk 2 above)
Comment (codeboten) [reply config/configauth/go.mod]: I moved the replace, ptal

**Review by codeboten — COMMENTED** [review]
Addressed all the newlines, ptal @jpkrohling

**Review by dmitryax — APPROVED** [review]

**Pull request merged**